cranelift/codegen/src/opts/bitops.isle

```diff
@@ -92,3 +92,17 @@
 (rule (simplify (bxor ty x (iconst ty k)))
   (if-let -1 (i64_sextend_imm64 ty k))
   (bnot ty x))
+
+;; bmask(input) = !(((input | ((!input) + 1)) >> 63) - 1)
```

Comment:
Nitpick: instead of hard coding `63` lets make this `bits(ty) - 1`.

Also, lets add a comment with an informal proof / argument about why this is correct, as @cfallin mentioned.

Reply:
I think the correctness argument is much less important to write out if this is split into the four separate rules I suggested. They're individually much easier to verify.